src/block_cache/block_chunkcache.c

```diff
@@ -184,75 +184,64 @@ __chunkcache_make_hash(WT_CHUNKCACHE *chunkcache, WT_CHUNKCACHE_HASHID *hash_id,
 }
 
 /*
- * __chunkcache_evict_one --
- *     Evict a single chunk from the chunk cache.
+ * __chunkcache_should_evict --
+ *     Decide if we can evict this chunk. In the current algorithm we only evict the chunks with the
+ *     zero access count. We always decrement the access count on the chunk that is given to us. The
+ *     thread accessing the chunk increments the access count. So we will only evict the chunks that
+ *     have not been accessed for a long time.
  */
-static void
-__chunkcache_evict_one(WT_SESSION_IMPL *session)
+static inline bool
+__chunkcache_should_evict(WT_CHUNKCACHE_CHUNK *chunk)
 {
-    WT_CHUNKCACHE *chunkcache;
-    WT_CHUNKCACHE_CHUNK *chunk_to_evict;
-    bool found_eviction_candidate;
-
-    chunkcache = &S2C(session)->chunkcache;
-    found_eviction_candidate = false;
-
-    /*
-     * 1. With the LRU list lock held, we remove the chunk at the list's tail and mark
-     *    that chunk as being evicted.
-     *    That prevents the code that removes outdated chunks from freeing the chunk before we do.
-     * 2. Remove the chunk from its chunk's chain, acquiring appropriate locks.
-     * 3. Free the chunk.
-     */
-    __wt_spin_lock(session, &chunkcache->chunkcache_lru_lock);
-    TAILQ_FOREACH_REVERSE(
-      chunk_to_evict, &chunkcache->chunkcache_lru_list, __wt_chunkcache_lru, next_lru_item)
-    {
-        if (chunk_to_evict->valid) {
-            TAILQ_REMOVE(&chunkcache->chunkcache_lru_list, chunk_to_evict, next_lru_item);
-            chunk_to_evict->being_evicted = true;
-            found_eviction_candidate = true;
-            break;
-        }
-    }
-    __wt_spin_unlock(session, &chunkcache->chunkcache_lru_lock);
-
-    if (!found_eviction_candidate)
-        return;
-
-    __wt_verbose(session, WT_VERB_CHUNKCACHE, "evict: %s(%u), offset=%" PRIu64 ", size=%" PRIu64,
-      (char *)&chunk_to_evict->hash_id.objectname, chunk_to_evict->hash_id.objectid,
-      (uint64_t)chunk_to_evict->chunk_offset, (uint64_t)chunk_to_evict->chunk_size);
-
-    __wt_spin_lock(session, WT_BUCKET_LOCK(chunkcache, chunk_to_evict->bucket_id));
-    TAILQ_REMOVE(
-      WT_BUCKET_CHUNKS(chunkcache, chunk_to_evict->bucket_id), chunk_to_evict, next_chunk);
-    __wt_spin_unlock(session, WT_BUCKET_LOCK(chunkcache, chunk_to_evict->bucket_id));
-    __chunkcache_free_chunk(session, chunk_to_evict);
-
-    WT_STAT_CONN_INCR(session, chunk_cache_chunks_evicted);
+    /* Do not evict chunks that are in the processed of being added to the cache. */
+    if (!chunk->valid)
+        return (false);
+    if (--(chunk->access_count) == 0)
+        return (true);
+    return (false);
 }
 
 /*
  * __chunkcache_eviction_thread --
- *     Periodically sweep the cache and evict chunks at the end of the LRU list.
+ *     Periodically sweep the cache and evict chunks with a low access count. This is similar to the
```

Comment:
Missed this during the review sorry. Nitpick: "low access count" -> "zero access count"